tests/native_tests/defclass.hy

```diff
@@ -109,3 +109,17 @@
   (assert (= a.y 2))
   (assert foo 2)
   (assert (.greet a) "hello"))
+
+(defn test-defclass-implicit-nil-for-init []
+  "NATIVE: test that defclass adds an implicit nil to --init--"
+  (defclass A []
+    [--init-- (fn [self] (setv self.x 1) 42)])
+  (defclass B []
+    (defn --init-- [self]
+      (setv self.x 2)
+      42))
```

Comment:
Well, if the return value wouldn't be nil, it'd blow up a few lines later, wouldn't it?